contracts/middleman/MerklGaugeMiddleman.sol

```diff
@@ -115,12 +109,23 @@ contract MerklGaugeMiddleman {
 
     /// @notice Transmits rewards from the `AngleDistributor` to the `DistributionCreator` with the correct
     /// parameters
-    /// @dev Only callable by the `AngleDistributor` contract
+    /// @dev Callable by any contract
+    /// @dev This method can be used to recover leftover ANGLE tokens in the contract
     function notifyReward(address gauge, uint256 amount) external {
         DistributionParameters memory params = gaugeParams[gauge];
-        if (msg.sender != angleDistributor() || params.uniV3Pool == address(0)) revert InvalidParams();
+        if (params.uniV3Pool == address(0)) revert InvalidParams();
+        if (amount == 0) amount = angle().balanceOf(address(this));
         params.epochStart = uint32(block.timestamp);
         params.amount = amount;
-        if (amount > 0) merklDistributionCreator().createDistribution(params);
+        DistributionCreator creator = merklDistributionCreator();
+        if (amount > 0) {
+            // Need to deal with minimum distribution amounts
+            if (amount > creator.rewardTokenMinAmounts(address(angle())) * params.numEpoch) {
+                merklDistributionCreator().createDistribution(params);
+            } else {
+                // Sending leftover ANGLE tokens to the `msg.sender`
+                angle().safeTransfer(msg.sender, amount);
```

Comment:
We shouldn't send leftover to the msg.sender as this can be anyone, so they can call the notifyReward with creator.rewardTokenMinAmounts(address(angle())) * params.numEpoch-1 and get back all the tokens sent by the AngleDistributor